app/src/main/java/com/willowtree/vocable/presets/PresetsFragment.kt

```diff
@@ -273,20 +228,69 @@ class PresetsFragment : BaseFragment() {
         }
     }
 
+    private fun handleCategories(categories: List<Category>) {
+        with(binding?.categoryView) {
+            this?.isSaveEnabled = false
+            this?.adapter = categoriesAdapter
+            categoriesAdapter.setItems(categories)
+
+            // The categories ViewPager position will initially be set to the middle so that the
+            // user can scroll in both directions. Upon subsequent config changes, the position
+            // will be set to the closest page to the middle which contains the selected category.
+            var targetPosition = categoriesAdapter.itemCount / 2
+
+            if (targetPosition % categoriesAdapter.numPages != 0) {
+                targetPosition %= categoriesAdapter.numPages
+            }
+
+            presetsViewModel.selectedCategory.value?.let { selectedCategory ->
+                for (i in targetPosition until targetPosition + categoriesAdapter.numPages) {
+                    val pageCategories = categoriesAdapter.getItemsByPosition(i)
+
+                    if (pageCategories.find { pageCategory -> pageCategory.categoryId == selectedCategory.categoryId } != null) {
```

Comment:
I think you can write this as:
```
pageCategories.firstOrNull { it.categoryId == selectedCategory.categoryId }?.let {
  targetPosition = i
  break
}
```

Reply:
hmm good suggestion but the break doesn't work in there. Wonder if there's a way to do it with return@label

Reply:
Yeah I've noticed that breaking from loops in Kotlin can be weird. I had to move a `return@` to the top of the loop for it to work in `LocaleUtils`